jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -1649,41 +1641,23 @@ private class HttpChannelSerializedInvoker extends SerializedInvoker
         protected void onError(Runnable task, Throwable failure)
         {
             ChannelRequest request;
-            Content.Chunk error;
             boolean callbackCompleted;
             try (AutoLock ignore = _lock.lock())
             {
                 callbackCompleted = _callbackCompleted;
                 request = _request;
-                error = _request == null ? null : _failure;
             }
 
             if (request == null || callbackCompleted)
             {
-                // It is too late to handle error, so just log it
+                // It is too late to handle error.
                 super.onError(task, failure);
+                return;
             }
-            else if (error == null)
-            {
-                // Try to fail the request, but we might lose a race.
-                try
-                {
-                    request._callback.failed(failure);
```

Comment:
I've thought about this some more and I'm OK with this failed called, at least until we are sure.  In the HandlerInvoker, we do similar if the call to handle throws.  This is much the same thing.

Reply:
@gregw not sure it's the same thing. If we throw from handle() we documented "equivalent to fail callback", but throwing e.g. from a request listener or from a demand callback or from a write callback does not seem equivalent as leaving handle() with an exception.

Reply:
@sbordet I'm not sure either.  Your build is green, so OK for this release... but it is a cautious OK.

Reply:
@sbordet throwing from a callback that is not meant to throw is a pretty good indication of an application that is not behaving correctly.   I wonder at the wisdom of letting it hang forever saying "you should not have thrown and should have called the callback yourself", when we already know it is broken and if they complain that we have failed the callback, then we just say "don't throw!"

Reply:
It might be better to always fail the callback in this case?

Reply:
I think of it in a different way: it's an unexpected failure and we have a mechanism for that, failure listeners.

Register a failure listener and you will be notified and you can complete the callback (as you can't really do much else).

[Here](https://github.com/jetty/jetty.project/issues/10916#issuecomment-1830166219) a proposal for a catch-all `Handler`.